fix(source-google-ads): clamp incremental start to retention window (#78504)
airbytehq/airbyte: docs/integrations/sources/google-ads.md

Streams that include metric fields (e.g., clicks, impressions, cost, conversions) may return fewer rows than dimension-only streams for the same resource. This is because the [Google Ads API omits rows where all metrics are zero](https://developers.google.com/google-ads/api/docs/reporting/zero-impressions) when metrics are included in the query. The omitted rows represent entity/segment combinations with no recorded activity.
:::

> For incremental streams, data is synced up to the previous day using your Google Ads account time zone since Google Ads can filter data only by [date](https://developers.google.com/google-ads/api/fields/v23/ad_group_ad#segments.date) without time. For report streams with daily or weekly segment data, Airbyte only queries data inside Google Ads' 37-month granular retention window; older data is skipped. If both `start_date` and `end_date` are older than the retention window, the connector checks availability using the earliest retained date and returns no historical records outside the retained window. If you use Overwrite sync mode, store historical data older than 37 months before upgrading to connector version 6.0.0 if you need to keep it. Existing older records in Append Dedup sync mode are preserved by that sync mode, but the connector no longer syncs them. Also, some reports cannot load data real-time due to Google Ads [limitations](https://support.google.com/google-ads/answer/2544985?hl=en).

Reviewer: ```suggestion
For incremental streams, data is synced up to the previous day using your Google Ads account time zone since Google Ads can filter data only by [date](https://developers.google.com/google-ads/api/fields/v23/ad_group_ad#segments.date) without time. For report streams with daily or weekly segment data, Airbyte only queries data inside Google Ads' 37-month granular retention window; older data is skipped. If both `start_date` and `end_date` are older than the retention window, the connector checks availability using the earliest retained date and returns no historical records outside the retained window. If you use the Overwrite sync mode, store historical data older than 37 months before upgrading to connector version 6.0.0 if you need to keep that historical data. Incremental | Append and Incremental | Append + Deduped sync modes preserve existing older records because of that sync mode, but the connector no longer syncs them. Also, some reports cannot load data real-time due to [Google Ads limitations](https://support.google.com/google-ads/answer/2544985?hl=en).
```

FROM ad_group
```

> Note that `segments.date` is automatically added to the `WHERE` clause if it is included in the `SELECT` clause. Custom reports including `segments.date` in the `SELECT` clause are synced by day for data inside Google Ads' 37-month granular retention window. Older report slices are skipped. If both `start_date` and `end_date` are older than the retention window, the connector checks availability using the earliest retained date and emits no rows for the configured historical date range. If the connection uses Overwrite sync mode, store any historical custom-query data older than 37 months before upgrading to connector version 6.0.0 if you need to keep it. Append Dedup sync mode preserves existing older records because of that sync mode, but the connector no longer emits them.

Reviewer: ```suggestion
Note that `segments.date` is automatically added to the `WHERE` clause if it is included in the `SELECT` clause. Custom reports including `segments.date` in the `SELECT` clause are synced by day for data inside Google Ads' 37-month granular retention window. Older report slices are skipped. If both `start_date` and `end_date` are older than the retention window, the connector checks availability using the earliest retained date and emits no rows for the configured historical date range. If the connection uses the Full Refresh | Overwrite sync mode, store historical custom-query data older than 37 months before upgrading to connector version 6.0.0 if you need to keep it. Incremental | Append and Incremental | Append + Deduped sync modes preserve existing older records because of that sync mode, but the connector no longer syncs them.
```

| Version     | Date       | Pull Request                                             | Subject                                                                                                                                                                |
|:------------|:-----------|:---------------------------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 5.0.2 | 2026-05-29 | [78508](https://github.com/airbytehq/airbyte/pull/78508) | Remove the Google Ads 400 response filter predicate to avoid buffering large streaming responses. |

> | 6.0.0 | 2026-05-29 | [78504](https://github.com/airbytehq/airbyte/pull/78504) | Clamp incremental report dates to Google Ads' 37-month granular data retention window. |

Reviewer: ```suggestion
| 6.0.0 | 2026-05-29 | [78504](https://github.com/airbytehq/airbyte/pull/78504) | Clamp incremental report dates to Google Ads' 37-month granular data retention window. |
| 5.0.2 | 2026-05-29 | [78508](https://github.com/airbytehq/airbyte/pull/78508) | Remove the Google Ads 400 response filter predicate to avoid buffering large streaming responses. |
```
